### PR TITLE
ChainerX test tool for ops

### LINKED ISSUE #6163: Unified test tool for ChainerX routines
We need a unified test tool for ChainerX routines, like done for Chainer functions in #3499. To minimize the implementation cost, it's good to reuse the existing facilities for Chainer Variable/FunctionNode (i.e., `chainer.gradient_check` and `FunctionTestCase`). We need the following:

- A tool to wrap a ChainerX routine function with `FunctionNode`
- Test case base class that combines this tool with `FunctionTestCase`
- Test case decorator for parameterizing the test cases with ChainerX backends

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #6163

### CODE REVIEW

File: chainer/testing/function.py [hunk 1]
```diff
@@ -218,8 +107,8 @@ def _forward(self, inputs, backend_config):
         _check_variable_types(outputs, backend_config.device, 'forward')
         return outputs
 
-    def test_forward(self, backend_config):
-        """Tests forward computation."""
+    def run_test_forward(self, backend_config):
```
Comment (asi1024) [inline chainer/testing/function.py]: How about leaving `test_forward` without erasing?

File: chainer/testing/function.py [hunk 2]
```diff
@@ -418,21 +433,27 @@ def _check_forward_output_arrays_equal(
                 enumerate(zip(expected_arrays, actual_arrays))):
             try:
                 array_module.assert_allclose(expected, actual, **opts)
-            except AssertionError:
-                indices.append(i)
+            except AssertionError as e:
+                indices.append((i, e))
```
Comment (asi1024) [inline chainer/testing/function.py]: The variable `indices` is not a list of indices, but an association list of errors.

Comment (niboshi) [reply chainer/testing/function.py]: Removed this change from this PR.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**